### ISSUE TITLE
Enforce matching nixpkgs and home-manager versions

### ISSUE THREAD

**Issue by danth** [danth/stylix#379]
Using Stylix with mismatched versions Nixpkgs or Home Manager is a common cause of one of the following errors:

- Missing options since they were implemented in a newer release
- Failing patches since we target a specific version of a package

See #375 and #241 for example, and many more questions on the [Matrix channel](https://matrix.to/#/#stylix:danth.me).

It shouldn't be too difficult to add a check against `config.system.nixos.release`, and its Home Manager counterpart. We can then ask the user to upgrade NixOS/downgrade Stylix rather than allowing it to fall through to a cryptic error.

**Comment by trueNAHO** [comment]
What about adding a warning in the documentation?

**Comment by danth** [comment]
That would also be helpful, although a check in code will also catch cases where someone didn't read the docs.

**Comment by trueNAHO** [comment]
Adding testbeds for every application (https://github.com/danth/stylix/issues/319) would avoid Stylix supporting upstream options that are not yet available through our `flake.lock`. This would simplify the process of figuring out when to update `flake.lock`.

**Comment by danth** [comment]
I agree, the version stored in our `flake.lock` should be guaranteed to work perfectly.

We still need to support people using slightly newer / older revisions to some extent, since updating nixpkgs regularly is important for security updates - so it's not reasonable to expect people to stay pinned to Stylix's lock file.

**Comment by trueNAHO** [comment]
> We still need to support people using slightly newer / older revisions to some extent, since updating nixpkgs regularly is important for security updates - so it's not reasonable to expect people to stay pinned to Stylix's lock file.

What about periodic updates? 

For example, I use [1] in my `dotfiles`. I could PR this, if you want.

[1]: https://github.com/trueNAHO/dotfiles/blob/dc4df5f53d9e5bc6dc10d4e44541af1a77148e84/.github/workflows/nix_flake_update.yml

**Comment by danth** [comment]
AFAIK that action doesn't trigger the GitHub Actions checks, since events caused by an Actions workflow can't trigger another workflow, so reviewing it's PRs would be inconvenient.

**Comment by Nebucatnetzer** [comment]
> What about adding a warning in the documentation?

That would definitely help or use a versioned example in the documentation.
I don't think it can be expected that people are using unstable by default.
